catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -106,13 +115,86 @@ public Object parseXml(InputStream xml) throws ValidationException {
     }
   }
 
-  private Geometry convertCRS(Geometry geometry) throws ValidationException, TransformException {
-    return JTS.transform(geometry, getLatLonTransform());
+  /**
+   * Iterate over the geometry collection and convert the non-standard LinearRing to LineString
+   *
+   * @param geometryCollection the geometryCollection to be iterated
+   * @return an edited geometryCollection with LineString in place of LinearRing
+   */
+  private GeometryCollection standardizeGeometryCollection(GeometryCollection geometryCollection) {
+    ArrayList<Geometry> geometries = new ArrayList<>();
+
+    for (int i = 0; i < geometryCollection.getNumGeometries(); i++) {
+      Geometry geometry = standardizeGeometry(geometryCollection.getGeometryN(i));
+      geometries.add(geometry);
+    }
+
+    return GEOMETRY_FACTORY.createGeometryCollection(geometries.toArray(new Geometry[0]));
+  }
+
+  /**
+   * Convert the non-standard LinearRing geometry to a LineString geometry
+   *
+   * @param geometry LinearRing to be converted
+   * @return new LineString geometry
+   */
+  private Geometry linearRingToLineString(Geometry geometry) {
+    return GEOMETRY_FACTORY.createLineString(((LinearRing) geometry).getCoordinateSequence());
+  }
+
+  /**
+   * Applies the math transformation to change Coordinate Reference Systems
+   *
+   * @param geometry Geometry to be transformed
+   * @return a Geometry with the transformation applied
+   * @throws ValidationException
+   * @throws TransformException
+   */
+  private Geometry applyMathTransform(Geometry geometry)
+      throws ValidationException, TransformException {
+    CoordinateReferenceSystem geometryCrs = getGeometryCrs(geometry);
+    return JTS.transform(geometry, getLatLonTransform(geometryCrs));
+  }
+
+  /**
+   * Standardizes the geometry by converting any instances of LinearRing to LineString.
+   *
+   * @param geometry the Geometry that may contain LinearRing
+   * @return a new Geometry with LineStrings in place of LinearRings
+   * @throws ValidationException
+   */
+  private Geometry standardizeGeometry(Geometry geometry) {
+    if (geometry instanceof GeometryCollection && !(geometry instanceof MultiPolygon)) {
```

Comment:
❓ Why only MultiPolygon? Do you want to check MutliPoint as well?

https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/GeometryCollection.html

Reply:
This was left behind from an attempt at using jts' GeometryEditor. I'll remove it.
In case you were curious:
I was using the GeometryEditor to change LinearRing to LinearString inside the GeometryCollection. The GeometryEditor wasn't only editing the Geometries in the GeometryCollection but also all embedded geometries as well. So a Polygon containing a valid LinearRing was also being changed to LineString and breaking stuff.
So I couldn't use the GeometryEditor and forgot to remove the MultiPolygon check, which didn't work anyway.